packages/zone/src/make-once.js

```diff
@@ -0,0 +1,70 @@
+// @ts-check
+const { Fail } = assert;
+
+/**
+ * @param {string} baseLabel
+ * @param {import('.').Stores} stores
+ */
+export const makeOnceKit = (baseLabel, stores) => {
+  const usedKeys = stores.detached().setStore(`${baseLabel} used keys`);
+
+  /**
+   * @template V
+   * @param {string} key
+   * @param {(key: string) => V} maker
+   * @returns {V}
+   */
+  const makeOnceUnvalidated = (key, maker) => {
```

Comment:
Where does this retrieve the existing value from the store (for the durable case)?

Reply:
Ok I guess I have a misunderstanding on the semantics. `makeOnce` here seem to be gating a `provide` callback, not a `maker` callback. Is that correct?

Reply:
That was my mistake.  I clarified now with `makeOnce` (which manages the durable backing store) and `wrapProvider` (which gates a `provide` callback).